index.ts

```diff
@@ -1,35 +1,52 @@
 export class OatyArray {
-  private _original: object[]
-  private _keys: string[]
-  private _transposed: object = {}
+  private _keys: string[] | undefined
+  private _original: object[] = []
+  private _transposed: { [key: string]: any } = {}
+
+  constructor(options?: { data?: object[], keys?: string[] }) {
+    if (options !== undefined) {
+      if (options.data !== undefined) { this.push(options.data) }
+      this._keys = options.keys
+    }
+  }
+
+  get keys(): string[] | undefined {
+    return this._keys
+  }
 
   get length(): number {
     return this._original.length
   }
 
-  constructor(input: object[], keys: string[]) {
-    this._original = input.map((x) => Object.assign({}, x))
-    this._keys = keys
+  get original(): object[] {
+    return this._original
+  }
 
-    this._transposed = transpose(this._keys, this._original)
+  get transposed(): object[] {
+    return this._original
   }
 
   public get(keyName: string, keyValue: string): object[] {

```

Comment:
Given not all data has to have all keys, it may be useful to make `keyValue` optional so that I could get all objects that have that key in O(1), too.